rs/genchains_rs.go

```diff
@@ -0,0 +1,121 @@
+//go:build ignore
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+	"unicode"
+
+	"github.com/smartcontractkit/chain-selectors/internal/gotmpl"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+const (
+	generatedFileName = "generated_chains.rs"
+	tmplFileName      = "generated_chains.rs.tmpl"
+)
+
+func wd() string {
+	rsDir := os.Getenv("PWD")
+	if !strings.HasSuffix(rsDir, "/rs") {
+		rsDir = path.Join(rsDir, "rs")
+	}
+	return rsDir
+}
+
+func main() {
+	rsDir := wd()
+	tmplRaw, err := os.ReadFile(path.Join(rsDir, tmplFileName))
+	if err != nil {
+		panic(err)
+	}
+	tmpl, err := template.New(generatedFileName).Parse(string(tmplRaw))
+	if err != nil {
+		panic(err)
+	}
+
+	generatedFilePath := path.Join(rsDir, "chainselectors", "src", generatedFileName)
+	existingContent, err := os.ReadFile(generatedFilePath)
+	if err != nil {
+		panic(err)
+	}
+
+	raw, err := gotmpl.Run(tmpl, newRustNameEncoder())
+	if err != nil {
+		panic(err)
+	}
+
+	formatted, err := rustfmt([]byte(raw))
+	if err != nil {
+		panic(err)
+	}
+
+	if string(existingContent) == string(formatted) {
+		fmt.Println("rust: no changes detected")
+		return
+	}
+
+	if err := os.WriteFile(generatedFilePath, formatted, 0644); err != nil {
+		panic(err)
+	}
+}
+
+func rustfmt(src []byte) ([]byte, error) {
+	tmpFile := path.Join(os.TempDir(), generatedFileName)
+	if err := os.WriteFile(tmpFile, src, 0644); err != nil {
+		return nil, err
+	}
+	defer os.Remove(tmpFile)
+
+	if err := exec.Command("rustfmt", tmpFile).Run(); err != nil {
+		// if rustfmt is not installed, try to use docker
+		vol := fmt.Sprintf("%s:/usr/src/app/%s", tmpFile, generatedFileName)
+		rustfmtCmd := fmt.Sprintf("rustup component add rustfmt &>/dev/null && rustfmt %s", generatedFileName)
+		if dockerErr := exec.Command(
+			"docker", "run", "--rm",
+			"-v", vol, "-w", "/usr/src/app",
+			"rust:1.82-alpine",
+			"/bin/sh", "-c", rustfmtCmd,
+		).Run(); dockerErr != nil {
+			return nil, err
+		}
+	}
+	formatted, err := os.ReadFile(tmpFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return formatted, nil
+}
```

Comment:
I'll make the fixes on the template but in general we can't rely on that and we must apply formatting.
Another thing is that we want to compare the generated file with the previous one, to ensure we have an updated version.. CI will fail if the rust code was not generated by mistake, similar to how it works for the Go code generation.

Reply:
I'm using docker as a fallback so others won't need to install rust toolchain to generate new chains, it will slow down the process on the first run but afterwards its pretty quick.